jme3-core/src/main/java/com/jme3/scene/shape/RectangleMesh.java

```diff
@@ -0,0 +1,163 @@
+import com.jme3.math.Rectangle;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Mesh;
+import com.jme3.scene.VertexBuffer.Type;
+
+/**
+ * {@code RectangleMesh} is a rectangle in space similar to
+ * {@link com.jme3.scene.shape.Quad}. It uses a
+ * {@link com.jme3.math.Rectangle} to position its vertices.
+ * 
+ * @author Francivan Bezerra
+ */
+public class RectangleMesh extends Mesh {
+
+    private Rectangle rectangle;
+
+    public RectangleMesh() {
+        rectangle = new Rectangle();
+    }
+
+    /**
+     * Creates a mesh with the given rectangle.
+     * 
+     * @param rectangle the rectangle used to build this mesh.
+     */
+    public RectangleMesh(Rectangle rectangle) {
+        this.rectangle = rectangle;
+        updateMesh();
+    }
+
+    /**
+     * Creates a mesh where points A, B and C define the area of the rectangle.
+     * 
+     * @param a the first corner of the rectangle.
+     * @param b the second corner of the rectangle.
+     * @param c the third corner of the rectangle.
+     */
+    public RectangleMesh(Vector3f a, Vector3f b, Vector3f c) {
+        this(new Rectangle(a, b, c));
+    }
+
+    /**
+     * Returns the rectangle used to build this mesh.
+     * 
+     * @return the rectangle used to build this mesh.
+     */
+    public Rectangle getRectangle() {
+        return rectangle;
+    }
+
+    /**
+     * Sets the rectangle that will be used to build this mesh.
+     * 
+     * @param rectangle the rectangle used to build this mesh.
+     */
+    public void setRectangle(Rectangle rectangle) {
+        this.rectangle = rectangle;
+        updateMesh();
+    }
+
+    public void updateMesh() {
+        updateMesh(false);
+    }
+
+    public void updateMesh(boolean flipCoords) {
+        // the fourth point defining the rectangle (C - B) + A.
+        final Vector3f fourthPoint = rectangle.getC().subtract(rectangle.getB()).addLocal(rectangle.getA());
+
+        setBuffer(Type.Position, 3,
+                new float[] {
+                        rectangle.getA().x, rectangle.getA().y, rectangle.getA().z,
+                        rectangle.getB().x, rectangle.getB().y, rectangle.getB().z,
+                        rectangle.getC().x, rectangle.getC().y, rectangle.getC().z,
+                        fourthPoint.x, fourthPoint.y, fourthPoint.z
+                });
+
+        if (flipCoords) {
+            setBuffer(Type.TexCoord, 2, new float[] {
+                    0, 1,
+                    1, 1,
+                    1, 0,
+                    0, 0
+            });
+        } else {
+            setBuffer(Type.TexCoord, 2, new float[] {
+                    0, 0,
+                    1, 0,
+                    1, 1,
+                    0, 1
+            });
+        }
+
+        setBuffer(Type.Normal, 3, new float[] {
+                0, 0,
+                1, 0,
+                0, 1,
+                0, 0,
+                1, 0,
+                0, 1
+        });
```

Comment:
These normals are only correct for a rectangle in the X-Y plane. Calculate the normal vector using the `com.jme3.math.Triangle` class or similar logic.

Reply:
I used the static method `Triangle.computeTriangleNormal()` to fix it.